File: 1432. Max Difference You Can Get From Changing an Integer/maxDiff.js
/**
 * @param {number} num
 * @return {number}
 */
var maxDiff = function (num) {
    let large = num
    let small = num

    let str = String(num)

    //create the SMALL number can be formed from num
    if (str[0] === "1") {
        for (let i = 1; i < str.length; i++) {
            if (str[i] !== "0" && str[i] !== "1") {
                small = parseInt(str.replaceAll(str[i], '0'))
                break;
            }
        }
    }
    else {
        small = parseInt(str.replaceAll(str[0], '1'))
    }


    //create the LARGE number can be formed from num
    for (let chr of str) {
        if (chr !== "9") {
            large = parseInt(str.replaceAll(chr, '9'))
            break
        }
    }

    return large - small
};

File: 1432. Max Difference You Can Get From Changing an Integer/maxDiff.cpp
#include <iostream>
#include <algorithm>

using namespace std;

class Solution {
    public:
    
        string replaceAll(string& str , char from, char to){
            for(auto &chr : str){
                if(chr == from) chr = to;
            }
            return str;
        }
    
        int maxDiff(int num) {
            string large = to_string(num);
            string small = to_string(num);
    
            //create SMALL number
            if(small[0] == '1'){
                for(int i=1 ; i<small.size(); i++){
                    if(small[i] != '0' && small[i] != '1'){
                        replaceAll(small, small[i] , '0');
                        break;
                    }
                }
            }
            else{
                replaceAll(small, small[0], '1');
            }
    
    
            //create LARGE number
            for(auto chr : large){
                if(chr != '9'){
                   replaceAll(large, chr , '9');
                   break;
                }
            }
            return stoi(large) - stoi(small);
        }
    };